pkg/declarativeconfig/transform/auth_provider.go

```diff
@@ -0,0 +1,151 @@
+package transform
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/auth/authproviders/iap"
+	"github.com/stackrox/rox/pkg/auth/authproviders/oidc"
+	"github.com/stackrox/rox/pkg/auth/authproviders/openshift"
+	"github.com/stackrox/rox/pkg/auth/authproviders/saml"
+	"github.com/stackrox/rox/pkg/auth/authproviders/userpki"
+	"github.com/stackrox/rox/pkg/declarativeconfig"
+	"github.com/stackrox/rox/pkg/errox"
+)
+
+var _ Transformer = (*authProviderTransform)(nil)
+
+type authProviderTransform struct{}
+
+func newAuthProviderTransformer() *authProviderTransform {
+	return &authProviderTransform{}
+}
+
+func (a *authProviderTransform) Transform(configuration declarativeconfig.Configuration) (map[reflect.Type][]proto.Message, error) {
+	authProviderConfig, ok := configuration.(*declarativeconfig.AuthProvider)
+	if !ok {
+		return nil, errox.InvalidArgs.Newf("invalid configuration type received for auth provider: %T", configuration)
+	}
+	authProviderProto := &storage.AuthProvider{
+		Id:                 declarativeconfig.NewDeclarativeAuthProviderUUID(authProviderConfig.Name).String(),
+		Name:               authProviderConfig.Name,
+		Type:               getType(authProviderConfig),
+		UiEndpoint:         authProviderConfig.UIEndpoint,
+		Enabled:            true,
+		Config:             getConfig(authProviderConfig),
+		LoginUrl:           "/sso/login/" + declarativeconfig.NewDeclarativeAuthProviderUUID(authProviderConfig.Name).String(),
```

Comment:
Hmm, yeah I wasn't sure at first whether this will be re-used, I think we can leave out setting it the custom way and rely on the option.
I'll add a comment though, in case we do need to revisit it and chose a different way for storing things.